drivers/counter/counter_esp32_rtc.c

```diff
@@ -107,8 +107,13 @@ static int counter_esp32_set_alarm(const struct device *dev, uint8_t chan_id,
 
 	counter_esp32_get_value(dev, &now);
 
+	if ((alarm_cfg->flags & COUNTER_ALARM_CFG_ABSOLUTE) == 0) {
+		WRITE_PERI_REG(RTC_CNTL_SLP_TIMER0_REG, (now + alarm_cfg->ticks));
+	} else {
+		WRITE_PERI_REG(RTC_CNTL_SLP_TIMER0_REG, alarm_cfg->ticks);
```

Comment:
Very nit question,are there not hal like functions equivalent to this?

Reply:
No, `hal_espressif` also uses the same approach. The only thing that is not implemented is supporting `uint64_t` for ticks value. But this requires Zephyr's API change.